### ISSUE TITLE
iOS18 Files.app issues

### ISSUE THREAD

**Issue by carloscabanero** [blinksh/blink#2051]
Some changes under the hood are affecting the old API experience:
- Opening a folder and then switching to a different location using the sidebar will "hang" (the other location will open +15 seconds after). Confirmed this still worked on 17.
- Uploading files to another app (like attaching files from Outlook), will hang. It requires to open the file locally or copying to a local location and then attaching to the app. This still works on 17.
- Contextual menu will sometimes not load. Intermittent issue.

This should be fixed as part of #1717

**Comment by prairie-guy** [comment]
Hi Carlos - Saw the new 18.0 update. From the notes, it doesn’t appear the Files.app bug was fixed. Moreover, still unable to use Files.app, so I’m assuming this its still an Open Issue. With iPadPro being my primary device, this is an important one for my day to day use. Any estimate on a fix?  Appreciate it.

**Comment by carloscabanero** [comment]
This is right now the number one priority. Good news is we already had the new API up and running, but there were a few improvements planned since WWDC. Plan is I will release a fix on TF in the next few days, and after a more thorough testing, it should be in production in the next two weeks.

In the meantime, what parts are broken for you? I may know a few ways to workaround them.

**Comment by mwiget** [comment]
I'm also facing this issue running 18.1 latest developer beta7. I had to remove file sharing from hosts in blink to avoid having Files app hanging for other apps, once I tried accessing a file via Blink. Files can see folders and files on a host but clicking on a file doesn't do anything visible and the cloud icon on the right remains. Hope this helps.

As a workaround I use SMB file access, as I also use my ipad pro as primary working device.

**Comment by carloscabanero** [comment]
For anyone having issues, if interested, drop me a DM on Discord and I will add you to TF to test the new FP implementation. I'm a bit behind but this is coming out soon.

**Comment by carloscabanero** [comment]
Update on this as I'm a bit behind. This is still the number one priority and working on it full time. It is taking me longer than I planned because the Provider behaves differently in corner cases (where it is trying to compose operations in order to create the flow it needs), and the only way to test this is through the extension itself, see how it behaves, what errors provoke what to the extension, etc...

I feel now confident about the file operations, so I will try to cut a TF version just focused on that. I think it will at least solve the urgency. Then I will continue refining the folder operations (moves, merges, etc...), as those are a bit more complex.

**Comment by prairie-guy** [comment]
I may have missed the fix, but I’m still unable to properly use Files.app with Blink. It looked like 18.1 was addressing this. I’ve downloaded it, reset config/hosts/Add_Location. The foreign FS loads, but I still can’t open files. If there is a fix on TF, could I please be added. Thanks

**Comment by carloscabanero** [comment]
We are waiting for 18.1 to be approved on TestFlight, hopefully they will approve today. Send me a DM on Discord and I will add you to it 👍🏼
